code/modules/clothing/shoes/magboots.dm

```diff
@@ -57,8 +57,31 @@
 		A.UpdateButtons()
 	check_mag_pulse(user)
 
+/obj/item/clothing/shoes/magboots/dropped(mob/user)
+	..()
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/H = user
+	if(H.get_item_by_slot(SLOT_HUD_SHOES) == src)
+		if(magpulse)
+			to_chat(user, "<span class='notice'>As [src] are removed, they deactivate.</span>")
+			attack_self(user, TRUE)
```

Comment:
You should never call `attack_self` directly

Reply:
not the same, i dont get why, but when equipping magboots it falls into this if -> if(H.get_item_by_slot(SLOT_HUD_SHOES) != src)

Reply:
> You should never call `attack_self` directly

just copied gravboots part of code, dont really know how to do it the other way